### PR TITLE
EVG-6456 allow specifying login domain

### PR DESCRIPTION
**Pull request by john-m-liu**
- Allow configuration of the domain of the login cookie
- Fix an unrelated bug where the patch UI routes don't send CORS headers if not authenticated (no functional difference, but you get a less weird error)

### CODE REVIEW

**Comment by ybrill** [comment]
What does adding a domain to the cookie enable?

**Comment by john-m-liu** [comment]
It allows the cookie to be shared between both evergreen.mongodb.com and spruce.mongodb.com

**Comment by ybrill** [comment]
by specifying `*.mongodb.com`?

**Comment by john-m-liu** [comment]
Yep. I'm hoping to do this so that users don't need to reauthenticate when they use spruce, though we'll have an actual auth system in the future

File: auth/github.go [hunk 1]
```diff
@@ -142,7 +150,7 @@ func (gum *GithubUserManager) GetLoginCallbackHandler() http.HandlerFunc {
 			grip.Errorf("Error sending code and authentication info to Github: %+v", err)
 			return
 		}
-		SetLoginToken(githubResponse.AccessToken, w)
+		SetLoginToken(githubResponse.AccessToken, gum.LoginDomain, w)
```
Comment (ybrill) [inline auth/github.go]: why is this only necessary for the GitHub user manager?

Comment (john-m-liu) [reply auth/github.go]: I'm just adding the ability to store a cookie with a configurable domain in all places that store the session cookie. I'm not actually sure this does anything, but nothing uses github auth anyway

**Pull request merged**